File: sem3/lab1/airport.cpp
#include <iostream>
#include <string>
#include <limits>
#include <locale>    
#include <codecvt> 


using namespace std;

class Airport {
    private:
        string airportName;
        int ticketPrice;
        int numberOfSeats;
        int ticketsSold;

    public:
        int priceOfSoldTitckes() {
            return ticketPrice * ticketsSold;
        }

        int getTicketsPrice() {
            return ticketPrice;
        }

        int getTicketsSold() {
            return ticketsSold;
        }   

        int getNumberOfSeats() {
            return numberOfSeats;
        }

        string getAirportName() {
            return airportName;
        }

        void setTicketsSold(int sold) {
            ticketsSold = sold;
        }

        void setTicketPrice(int price) {
            ticketPrice = price;
        }

        void setNumberOfSeats(int seats) {
            numberOfSeats = seats;
        }

        void setAirportName(string name) {
            airportName = name;
        }

};

int getIntInput(string prompt, string errorMessage,int minValue = numeric_limits<int>::min() , int maxValue = numeric_limits<int>::max()) {
    int value;
    cout << prompt;
    cin >> value;
    while(cin.fail() || maxValue < value || value < minValue) {
        cin.clear();
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
        cout << errorMessage << endl;
        cin >> value;
    }
    
    return value;
}

string getStringInput(string prompt, string errorMessage) {
    string value;
    cout << prompt;
    cin >> value;
    while(value.empty() || cin.fail()) {
        cin.clear();
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
        cout << errorMessage << endl;
        cin >> value;
    }
    
    return value;
}

int main(){
    setlocale(LC_ALL, "ru_RU.UTF-8");
    Airport Airport;
    Airport.setAirportName(getStringInput("Введите имя аэропорта: ", "Ошибка! Введите корректное имя аэропорта: "));
    Airport.setTicketPrice(getIntInput("Введите цену билета: ", "Ошибка! Введите корректную цену билета: ", 1));
    Airport.setNumberOfSeats(getIntInput("Введите количество мест: ", "Ошибка! Введите корректное количество мест: ", 1)); 
    Airport.setTicketsSold(getIntInput("Введите количество проданных билетов: ", "Ошибка! Введите корректное количество проданных билетов: ", 0, Airport.getNumberOfSeats()));
    cout << "------------------------" << endl;
    cout << "Имя аэропорта: " << Airport.getAirportName() << endl;
    cout << "Цена билета: " << Airport.getTicketsPrice() << endl;
    cout << "Количество мест: " << Airport.getNumberOfSeats() << endl;
    cout << "Количество проданных билетов: " << Airport.getTicketsSold() << endl;
    cout << "------------------------" << endl;
    cout << "Цена проданных билетов: " << Airport.priceOfSoldTitckes() << endl;
    return 0;
}

File: sem3/lab1/airport.py
class Airport:

    def __init__(self, airportName: str, ticketPrice: int, numberOfSeats: int, ticketsSold: int):
        self.airportName = airportName
        self.ticketPrice = ticketPrice
        self.numberOfSeats = numberOfSeats
        self.ticketsSold = ticketsSold

    def getAirportName(self) -> str:
        return self.airportName

    def getTicketPrice(self) -> int:
        return self.ticketPrice

    def getNumberOfSeats(self) -> int:
        return self.numberOfSeats

    def getTicketsSold(self) -> int:
        return self.ticketsSold
    
    def setAirportName(self, airportName: str):
        self.airportName = airportName
    
    def setTicketPrice(self, ticketPrice: int):
        self.ticketPrice = ticketPrice  
    
    def setNumberOfSeats(self, numberOfSeats: int):
        self.numberOfSeats = numberOfSeats
    
    def setTicketsSold(self, ticketsSold: int):
        self.ticketsSold = ticketsSold  
    
    def priceOfSoldTickets(self) -> int:
        return self.ticketPrice * self.ticketsSold


def main():
    while True:
        airportName = input("Введите имя аэропорта: ")
        if airportName.strip():
            break
        print("Ошибка! Введите корректное имя аэропорта.")

    while True:
        try:
            ticketPrice = int(input("Введите цену билета: "))
            if ticketPrice > 0:
                break
            else:
                print("Ошибка! Введите корректную цену билета.")
        except ValueError:
            print("Ошибка! Введите корректную цену билета.")

    while True:
        try:
            numberOfSeats = int(input("Введите количество мест: "))
            if numberOfSeats > 0:
                break
            else:
                print("Ошибка! Введите корректное количество мест.")
        except ValueError:
            print("Ошибка! Введите корректное количество мест.")

    while True:
        try:
            ticketsSold = int(input("Введите количество проданных билетов: "))
            if 0 <= ticketsSold <= numberOfSeats:
                break
            else:
                print("Ошибка! Введите корректное количество проданных билетов.")
        except ValueError:
            print("Ошибка! Введите корректное количество проданных билетов.")  

    airport = Airport(airportName=airportName, ticketPrice=ticketPrice, numberOfSeats=numberOfSeats, ticketsSold=ticketsSold)
    print("------------------------")
    print("Имя аэропорта:", airport.getAirportName())
    print("Цена билета:", airport.getTicketPrice())
    print("Количество мест:", airport.getNumberOfSeats())
    print("Количество проданных билетов:", airport.getTicketsSold())
    print("------------------------")
    print("Цена проданных билетов:", airport.priceOfSoldTickets())


if __name__ == "__main__":
    main()